exasol_script_languages_container_ci_setup/lib/run_start_release_build.py

```diff
@@ -0,0 +1,95 @@
+    :raises:
+        `RuntimeError` if build goes wrong or if anything on AWS CodeBuild is not as expected
+        `ValueError` if project is not found on AWS CodeBuild or if the upload is not in expected format.
+    The upload url is only be used to get the release id.
+    """
+    logging.info(f"run_start_release_build for aws profile {aws_access.aws_profile} for project {project} "
+                 f"with upload url: {upload_url}")
+    resources = aws_access.get_all_stack_resources(release_stack_name(project))
+    matching_project = [resource for resource in resources if resource["ResourceType"] == "AWS::CodeBuild::Project"]
+    if len(matching_project) == 0:
+        raise ValueError(f"No project deployed for {project}. Found following resources: {resources}")
+    if len(matching_project) > 1:
+        raise RuntimeError(f"Multiple projects match {project}. Found following matches: {matching_project}")
+    dry_run_value = "--no-dry-run"
+
+    gh_token = os.getenv("GITHUB_TOKEN")
+    if gh_token is None:
+        raise RuntimeError("Environment variable GITHUB_TOKEN needs to be declared.")
+    env_variables = [("RELEASE_ID", f"{_parse_upload_url(upload_url=upload_url)}"),
+                     ("DRY_RUN", dry_run_value),
+                     ("GITHUB_TOKEN", gh_token)]
+    environment_variables_overrides = list(map(get_environment_variable_override, env_variables))
+    aws_access.start_codebuild(matching_project[0]["PhysicalResourceId"],
+                               environment_variables_overrides=environment_variables_overrides,
+                               branch=branch)
+
+
+def run_start_test_release_build(aws_access: AwsAccess, gh_release_creator: GithubReleaseCreator, repo_id: str,
+                                 project: str, branch: str, release_title: str) -> None:
+    """
+    This command executes
+    1. Create the Github draft release for the given branch and title.
+    2. Retrieve resources for the release codebuild stack for that given project
+    3. Find the resource with type CodeBuild
+    4. Creates the environment variables override
+    5. Start and wait for batch build
+    :raises:
+        `RuntimeError` if build goes wrong or if anything on AWS CodeBuild is not as expected
+        `ValueError` if project is not found on AWS CodeBuild or if the upload is not in expected format.
+    The tag is used to declare the branch and to declare the release key.
+    """
+    logging.info(f"run_start_test_release_build for aws profile {aws_access.aws_profile} for project {project} "
+                 f"for branch: {branch} with title: {release_title}")
+    release_id = gh_release_creator.create_release(repo_id, branch, release_title)
+    resources = aws_access.get_all_stack_resources(release_stack_name(project))
+    matching_project = [resource for resource in resources if resource["ResourceType"] == "AWS::CodeBuild::Project"]
+    if len(matching_project) == 0:
```

Comment:
If possible refactor all common parts from start test release and start release, such that we have the maximum amount of code under test